### PR TITLE
Range graph test

### PR DESCRIPTION
**Pull request by mjkkirschner**
Screen Shot 2022-07-18 at 2 59 30 PM

### Purpose

* Adds a graph test for a double range passed to a function that takes an argument of `double[]`.
* improves workspace comparison constructor so it can handle arrays from the MSIL engine.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/ProtoCore/Utils/CoreUtils.cs [hunk 1]
```diff
@@ -892,6 +894,26 @@ public static object GetDataOfValue(Mirror.MirrorData value)
...
 
+        public static object GetDataOfCLRValue(object value)
+        {
+                // This is because stackvalues directly store value types
+                // while they store the string representation of reference types.
+                if (value is long || value is int || value is double || value is char || value is bool)
+                {
+                    return value;
+                }
+                else if (value is DesignScript.Builtin.Dictionary)
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Utils/CoreUtils.cs:905]: formatting?

File: src/Engine/ProtoCore/Utils/CoreUtils.cs [hunk 2]
```diff
@@ -892,6 +894,26 @@ public static object GetDataOfValue(Mirror.MirrorData value)
...
+            {
+                throw new NotImplementedException("get clr value of DSDictionary.");
+            }
+                //TODO expand.
+                else if(value is ICollection vc)
+            {
+                return vc.Cast<Object>().Select(x => GetDataOfCLRValue(x)).ToList();
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Utils/CoreUtils.cs:912]: 👍

**Review by aparajit-pratap — APPROVED** [review]

File: src/Engine/ProtoCore/Utils/CoreUtils.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Engine/ProtoCore/Utils/CoreUtils.cs:905]: will fix

File: src/Engine/ProtoCore/Utils/CoreUtils.cs (hunk 2 above)
Comment (pinzart90) [reply src/Engine/ProtoCore/Utils/CoreUtils.cs:912]: Is `ICollection` the only interface that allows iterations that might be found here ?
Or should we use IEnumerable ?

Comment (mjkkirschner) [reply src/Engine/ProtoCore/Utils/CoreUtils.cs:912]: I think right now, we always return an `IList` so thats an ICollection, and array also implement ICollection:
https://docs.microsoft.com/en-us/dotnet/api/system.array?view=net-6.0

I think String and Dictionary implement IEnumerable, but I guess as long as we check for string and Dictionary first it's safe.

My thought was to start small and expand this check as our graph test suite expands, thus the `TODO`.

**Pull request merged**